src/software/simulated_tests/tbots_test_fixture.py

```diff
@@ -0,0 +1,174 @@
+import time
+
+import decorator
+
+import pytest
+from software.simulated_tests.pytest_main import load_command_line_arguments
+from software.simulated_tests.simulated_test_fixture import simulated_test_initializer
+from software.field_tests.field_test_fixture import field_test_initializer
+import threading
+from software.thunderscope.proto_unix_io import ProtoUnixIO
+from software.thunderscope.thunderscope import Thunderscope
+from software.logger.logger import createLogger
+
+simulator_proto_unix_io = None
+yellow_full_system_proto_unix_io = None
+blue_full_system_proto_unix_io = None
+
+logger = createLogger(__name__)
+
+LAUNCH_DELAY_S = 0.1
+PAUSE_AFTER_FAIL_DELAY_S = 2
+TEST_END_DELAY = 0.3
+
+
+def initialize_unix_io():
+    """
+    Initializes proto unix sockets in global scope so that they are accessible from within tests
+
+    """
+    global simulator_proto_unix_io
+    global yellow_full_system_proto_unix_io
+    global blue_full_system_proto_unix_io
+
+    simulator_proto_unix_io = ProtoUnixIO()
+    yellow_full_system_proto_unix_io = ProtoUnixIO()
+    blue_full_system_proto_unix_io = ProtoUnixIO()
+
+
+def enable_thunderscope(test):
```

Comment:
I thought about this and some other potential features we may want a bit more, and I was wondering about what your thoughts are about making this a pytest fixture? Then the 3 pytest fixtures can depend on this one, so it is ran before every test using a one of the test fixtures without a need for a decorator. Another main reason I was thinking about this was to enable us to use `@pytest.fixture(scope="session")` so Thunderscope is launched once when running multiple simulated tests (rather than it opening and closing for every test). Though we can do that in a separate PR as there will probably need to be some work done for resetting the field widget and the other widgets between each test.